### PR TITLE
Fixed worker pool used for index lookups

### PR DESCRIPTION
**Pull request by bheni**
Current implementation can create a 100s of millions of go routines.  This creates a fixed pool of routines to do random access lookups.

### CODE REVIEW

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

**Comment by bheni** [comment]
#benchmark

File: go/libraries/doltcore/sqle/index_row_iter.go [hunk 1]
```diff
@@ -52,81 +59,86 @@ func NewIndexLookupRowIterAdapter(ctx *sql.Context, idx DoltIndex, keyIter nomsK
...
+	return nil, io.EOF
 }
 
-func (*indexLookupRowIterAdapter) Close() error {
+func (i *indexLookupRowIterAdapter) Close() error {
+	resultBufferPool.Put(i.resultBuf)
 	return nil
 }
 
 // queueRows reads each key from the key iterator and runs a goroutine for each logical processor to process the keys.
```
Comment (reltuk) [inline go/libraries/doltcore/sqle/index_row_iter.go:110]: comment wrong now...

**Review by reltuk — APPROVED** [review]
LGTM! One comment.

**Comment by bheni** [comment]
killing in favor of https://github.com/dolthub/dolt/pull/1304

**Pull request merged**